app/models/user.rb

```diff
@@ -80,8 +80,60 @@ class User < ApplicationRecord
 
   attribute :renewal_request, :string
 
-  attribute :created_at, :datetime, precision: 6
-  attribute :updated_at, :datetime, precision: 6
-
   validates :email, presence: true
+
+  # Finds, creates, or updates user from userinfo
+  # Find in case of user with existing token matching userinfo["sub"]
+  # Create in case of no token or email matching in userinfo
+  # Update in case of matching email to userinfo["email"] but no token set
+  # TODO: Add relevant security log tracking here?
+  def self.user_from_userinfo(userinfo)
+    email = userinfo[0]["email"]
+    token = userinfo[0]["sub"]
+
+    if (user = find_by(token:))
+      user
+    elsif (user = find_by(email:))
+      update_admin_added_user(user, userinfo)
```

Comment:
I am assuming the token changes each time they login. then this method will be called any time the user's token changed but they already have an account for that email so I'd probably rename it something more general like
```suggestion
      update_user_token_for_email(user, userinfo)
```
but correct me if I'm wrong about the token changing